src/main/java/Duke/Duke.java

```diff
@@ -0,0 +1,167 @@
+            System.out.println((i + 1) + ". " + (tasks.get(i)).toString());
+        }
+        System.out.println("Now you have " + tasks.size() + " tasks in the list." + dividingLine);
+    }
+
+    public static void addTask(Task task) {
+        tasks.add(task);
+        System.out.println(dividingLine);
+        System.out.println("Got it. I've added this task:");
+        System.out.println(task);
+        System.out.println("Now you have " + tasks.size() + " tasks in the list.");
+        System.out.println(dividingLine);
+    }
+
+    public static void processTask(String line) throws DukeException, TaskInfoException, MarkIndexException, NoTaskException {
+        if (line.equals("list")) {
+            printList();
+        } else if (line.contains("mark") && !line.contains("unmark")) {
+            if (tasks.size() == 0) {
+                throw new NoTaskException();
+            }
+            if (line.length() == 4) {
+                throw new MarkIndexException();
+            }
+            int taskNum = Integer.parseInt(line.substring(5)) - 1;
+            tasks.get(taskNum).markAsDone();
+            System.out.println(dividingLine + "Nice! I've marked this task as done:");
+            System.out.println("[" + tasks.get(taskNum).getStatusIcon() + "] " + tasks.get(taskNum).description
+                    + dividingLine);
+        } else if (line.contains("delete")) {
+            if(line.length() == 6){
+                throw new MarkIndexException();
+            }
+            int taskNum = Integer.parseInt(line.substring((7))) - 1;
+            System.out.println(dividingLine + "Noted. I've removed this task:");
+            System.out.println(tasks.get(taskNum));
+            System.out.println("Now you have " + (tasks.size() - 1) + " tasks in the list.");
+            System.out.println(dividingLine);
+            tasks.remove(taskNum);
+        } else if (line.contains("unmark")) {
+            if (tasks.size() == 0) {
+                throw new NoTaskException();
+            }
+            if (line.length() == 6) {
+                throw new MarkIndexException();
+            }
+            int unmarkNum = Integer.parseInt(line.substring(7)) - 1;
+            tasks.get(unmarkNum).unmarkAsDone();
+            System.out.println(dividingLine + "OK, I've marked this task as not done yet:");
+            System.out.println("[" + tasks.get(unmarkNum).getStatusIcon() + "] " + tasks.get(unmarkNum).description
+                    + dividingLine);
+        } else if (line.contains("todo")) {
+            addTask(new Todo(line.substring(5)));
+        } else if (line.contains("deadline")) {
+            line = line.replace("deadline ", "");
+            if (line.substring(line.indexOf("/by ") + 4).length() == 0) {
+                throw new TaskInfoException();
+            }
+            addTask(new Deadline(line.substring(0, line.indexOf(" /")),
+                    line.substring(line.indexOf("/by ") + 4)));
```

Comment:
You can refactor magic literals into constants to improve readability. Look for similar issue in other places.